lib/plugins/aws/deploy/lib/uploadCloudFormationTemplate.js

```diff
@@ -0,0 +1,32 @@
+'use strict';
+
+const BbPromise = require('bluebird');
+
+module.exports = {
+  uploadCloudFormationTemplateToS3Bucket() {
+    this.serverless.cli.log('Uploading CloudFormation file to S3...');
+
+    const body = JSON.stringify(this.serverless.service.provider.compiledCloudFormationTemplate);
+
+    const params = {
+      Bucket: this.bucketName,
+      Key: 'compiled-cloudformation-template.json',
+      Body: body,
+    };
+
+    return this.sdk.request('S3',
+      'putObject',
+      params,
+      this.options.stage,
+      this.options.region);
+  },
+
+  uploadCloudFormationTemplate() {
```

Comment:
Directory. This could also solve the artifact issue if we don't just take
the artifact name but create one ourselves right away

Am 18.08.2016 18:38 schrieb "Philipp Muens" <notifications@github.com>:

> In lib/plugins/aws/deploy/lib/uploadCloudFormationTemplate.js
> <https://github.com/serverless/serverless/pull/1886#discussion_r75343240>:
>
> > +    const body = JSON.stringify(this.serverless.service.provider.compiledCloudFormationTemplate);
> > +
> > +    const params = {
> > +      Bucket: this.bucketName,
> > +      Key: 'compiled-cloudformation-template.json',
> > +      Body: body,
> > +    };
> > +
> > +    return this.sdk.request('S3',
> > +      'putObject',
> > +      params,
> > +      this.options.stage,
> > +      this.options.region);
> > +  },
> > +
> > +  uploadCloudFormationTemplate() {
>
> Sounds like a good idea.
> Do you really mean bucket or directory? Because creating a new bucket on
> each deployment would take very long (until everything is propagated). So I
> would vote to create a directory with the naming schema you mentioned and
> then just put all the resources inside of this directory.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/serverless/serverless/pull/1886/files/5519402c80fa10be40ce7867a98f47cf0bff211a#r75343240>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAIGLWtANj2Veb3GaVGZxjMJTMBZQ4Frks5qhIqdgaJpZM4Jmhf->
> .
>